### PR TITLE
[FIX] goDown on empty string should not have any effects on the iterator

### LINKED ISSUE #2072: goDown on empty string breaks iterator
typedef Index<DnaString, FMIndex<> > TIndex;
    typedef Iter<TIndex, VSTree<TopDown<> > > TIter;

    DnaString text("GCCTC");
    TIndex index(text);
    TIter it(index);

    goDown(it, "C");    cout << representative(it) << endl; // outputs C
    goDown(it);         cout << representative(it) << endl; // outputs CC
    //goDown(it, "");     cout << representative(it) << endl; // outputs CC
    goRight(it);        cout << representative(it) << endl; // outputs GC
    goRight(it);        cout << representative(it) << endl; // outputs TC

When you comment in the line above, ``TC`` is not found anymore when calling goRight from ``GC``.

This might also happen with other index types. Going down an empty string should not touch the iterator at all.

### PR DESCRIPTION
**Pull request by cpockrandt**
Fixes #2072

### CODE REVIEW

**Comment by cpockrandt** [comment]
@smehringer You got it! ✌️

File: include/seqan/index/index_bifm_stree.h [hunk 1]
```diff
@@ -86,6 +86,9 @@ _goDownString(Iter<Index<TText, BidirectionalIndex<FMIndex<TOccSpec, TIndexSpec>
     TStringIter stringIt = begin(string, Standard());
     TStringIter stringEnd = end(string, Standard());
 
+    if (stringIt == stringEnd)
```
Comment (h-2) [inline include/seqan/index/index_bifm_stree.h]: please add SEQAN_UNLIKELY

**Comment by h-2** [comment]
Now you introduced some whitespace changes. Either undo the changes or properly delete the double-newlines, please!

**Comment by cpockrandt** [comment]
Thanks for the hint!

**Pull request merged**